src/System.Private.CoreLib/shared/System/Threading/Mutex.Windows.cs

```diff
@@ -84,7 +68,7 @@ private static OpenExistingResult OpenExistingWorker(string name, out Mutex resu
                 if (errorCode == Interop.Errors.ERROR_FILENAME_EXCED_RANGE)
                 {
                     // On Unix, length validation is done by CoreCLR's PAL after converting to utf-8
-                    throw new ArgumentException(SR.Format(SR.Argument_WaitHandleNameTooLong, name, WaitHandleNameMax), nameof(name));
```

Comment:
What is this comment referring to? Is there some additional length validation in the pal that needs to be removed?

Reply:
This is referring to `SHARED_MEMORY_MAX_FILE_PATH_CHAR_COUNT`. I would look at it separately from this fix.